fs/glob.ts

```diff
@@ -100,22 +116,93 @@ export async function* expandGlob(
   globString: string,
   {
     root = cwd(),
+    exclude = [],
     includeDirs = true,
     extended = false,
     globstar = false,
-    strict = false,
-    filepath = true,
-    flags = ""
+    strict = false
   }: ExpandGlobOptions = {}
 ): AsyncIterableIterator<WalkInfo> {
-  const absoluteGlob = isAbsolute(globString)
-    ? globString
-    : join(root, globString);
-  const globOptions = { extended, globstar, strict, filepath, flags };
-  yield* walk(root, {
-    match: [glob(absoluteGlob, globOptions)],
-    includeDirs
-  });
+  const resolveFromRoot = (path: string): string =>
+    isAbsolute(path) ? normalize(path) : join(root, path);
+  const globOptions: GlobOptions = { extended, globstar, strict };
+  const excludePatterns = exclude
+    .map(resolveFromRoot)
+    .map((s: string): RegExp => globToRegExp(s, globOptions));
+  const shouldInclude = ({ filename }: WalkInfo): boolean =>
+    !excludePatterns.some((p: RegExp): boolean => !!filename.match(p));
```

Comment:
If you're not interested in matched object, you can use regexp.test() method.

`!!filename.match(p)` => `p.test(filename)`

Reply:
`String::match()` doesn't modify `RegExp::lastIndex`. Honestly it doesn't matter here because the exclude patterns don't have `g` but I think this'll be more stable.